pulpcore/cli/core/export.py

```diff
@@ -86,20 +80,21 @@ def list(ctx: click.Context, exporter: str, limit: int, offset: int, **kwargs: A
 @click.option("--chunk-size", type=str, help="Examples: 512MB, 1GB")
 @click.option("--versions", type=tuple([str, str, int]), multiple=True)
 @click.option("--start-versions", type=tuple([str, str, int]), multiple=True)
-@click.pass_context
+@pass_entity_context
+@pass_pulp_context
 def run(
-    ctx: click.Context,
+    pulp_ctx: PulpContext,
+    export_ctx: PulpExportContext,
     exporter: str,
     full: bool,
     chunk_size: str,
     versions: List[RepositoryVersionDefinition],
     start_versions: List[RepositoryVersionDefinition],
 ) -> None:
-    pulp_ctx: PulpContext = ctx.find_object(PulpContext)
-    pexport_ctx: PulpExportContext = ctx.find_object(PulpExportContext)
-    the_exporter = pexport_ctx.find_exporter(exporter)
+    exporter_ctx = PulpExporterContext(pulp_ctx)
+    exporter_href = exporter_ctx.find(name=exporter)["pulp_href"]
     params = {
-        pexport_ctx.HREF: the_exporter["pulp_href"],
+        export_ctx.HREF: exporter_href,  # TODO This looks WRONG!!!
```

Comment:
RE "LOOKS WRONG" - exporter-to-export is always a little odd - will try to dig up some time to experiment w/this today.

Reply:
Oh, almost forgot about this. I'm glad my comment triggered you.